### ISSUE TITLE
crash in autowarmth plugin

### ISSUE THREAD

**Issue by symphorien** [koreader/koreader#9434]
* KOReader version: 2022.07
* Device: Kobo aura

#### Issue

crash

#### Steps to reproduce

presumably use koreader between dust and night with "switch front light off during day" activated

##### `crash.log` (if applicable)

```
./luajit: plugins/autowarmth.koplugin/main.lua:310: attempt to compare nil with number
stack traceback:
	plugins/autowarmth.koplugin/main.lua:310: in function 'scheduleNextWarmthChange'
	plugins/autowarmth.koplugin/main.lua:374: in function 'action'
	frontend/ui/uimanager.lua:1197: in function '_checkTasks'
	frontend/ui/uimanager.lua:1627: in function 'handleInput'
	frontend/ui/uimanager.lua:1744: in function 'run'
	./reader.lua:333: in main chunk
	[C]: at 0x000140bd
!!!!
Uh oh, something went awry... (Crash n°1: 08/13/22 @ 21:41:18)
Running FW 4.12.12111 on Linux 3.0.35+ (#5030 PREEMPT Wed Oct 25 10:25:24 CST 2017)
```

<details>
  <summary>crash.log</summary>

```
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 08/13/22-16:15:22
 [*] Version: v2022.07

ffi.load: blitbuffer
08/13/22-16:15:23 INFO  initializing for device Kobo_star
08/13/22-16:15:23 INFO  framebuffer resolution: {
    ["w"] = 758,
    ["h"] = 1024,
}
[ko-input] Forked off fake event generator (pid: 17565)
08/13/22-16:15:23 WARN  Device:getKeyRepeat: EVIOCGREP ioctl failed: Function not implemented
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
ffi.load: libs/libutf8proc.so.2
08/13/22-16:15:25 INFO  Loading plugins from directory: plugins
ffi.load: sqlite3
ffi.load: libs/liblept.so.5
ffi.load: libs/libk2pdfopt.so.2
error: Unable to read ICC workflow
warning: Attempt to read Output Intent failed
error: ICC Profiles not supported in NO_ICC build
warning: could not load ICC profile in JPEG image
08/13/22-16:15:47 INFO  opening file /mnt/onboard/Jaworski, Jean-Philippe/Meme pas mort - Jean-Philippe Jaworski.epub
08/13/22-16:16:18 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-16:16:18 INFO  Inhibiting user input
08/13/22-16:16:33 INFO  Kobo suspend: going to sleep . . .
08/13/22-16:16:33 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-16:16:35 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-16:16:36 INFO  Kobo suspend: synced FS
08/13/22-16:16:36 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-16:34:37 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-16:34:37 INFO  Kobo resume: clean up after wakeup
08/13/22-16:34:37 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-16:34:37 INFO  Restoring user input handling
08/13/22-16:55:12 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-16:55:12 INFO  Inhibiting user input
08/13/22-16:55:28 INFO  Kobo suspend: going to sleep . . .
08/13/22-16:55:28 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-16:55:30 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-16:55:30 INFO  Kobo suspend: synced FS
08/13/22-16:55:30 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-16:59:22 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-16:59:22 INFO  Kobo resume: clean up after wakeup
08/13/22-16:59:22 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-16:59:22 INFO  Restoring user input handling
08/13/22-17:05:15 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-17:05:15 INFO  Inhibiting user input
08/13/22-17:05:30 INFO  Kobo suspend: going to sleep . . .
08/13/22-17:05:30 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-17:05:32 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-17:05:33 INFO  Kobo suspend: synced FS
08/13/22-17:05:33 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-17:07:33 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-17:07:33 INFO  Kobo resume: clean up after wakeup
08/13/22-17:07:33 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-17:07:33 INFO  Restoring user input handling
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
08/13/22-17:49:31 INFO  Kobo Wi-Fi: enabling Wi-Fi
dhcpcd[17731]: dhcpcd not running
dhcpcd[17738]: version 6.6.6 starting
dhcpcd[17738]: eth0: using /sys hwaddr: 58:b0:d4:35:64:fc
dhcpcd[17738]: eth0: executing `/libexec/dhcpcd-run-hooks' PREINIT
dhcpcd[17738]: eth0: executing `/libexec/dhcpcd-run-hooks' CARRIER
dhcpcd[17738]: eth0: delaying IPv4 for 0.7 seconds
dhcpcd[17738]: eth0: using ClientID 01:58:b0:d4:35:64:fc
dhcpcd[17738]: eth0: soliciting a DHCP lease
dhcpcd[17738]: eth0: sending DISCOVER (xid 0x5a253543), next in 3.2 seconds
dhcpcd[17738]: eth0: offered 192.168.1.11 from 192.168.1.1
dhcpcd[17738]: eth0: sending REQUEST (xid 0x5a253543), next in 4.3 seconds
dhcpcd[17738]: eth0: acknowledged 192.168.1.11 from 192.168.1.1
dhcpcd[17738]: eth0: leased 192.168.1.11 for 86400 seconds
dhcpcd[17738]: eth0: renew in 43200 seconds, rebind in 75600 seconds
dhcpcd[17738]: eth0: writing lease `/var/db/dhcpcd-eth0-Kiwi.lease'
dhcpcd[17738]: eth0: adding IP address 192.168.1.11/24
dhcpcd[17738]: eth0: adding route to 192.168.1.0/24
dhcpcd[17738]: eth0: adding default route via 192.168.1.1
dhcpcd[17738]: eth0: executing `/libexec/dhcpcd-run-hooks' BOUND
dhcpcd[17738]: forking to background
dhcpcd[17738]: forked to background, child pid 17755
08/13/22-17:49:39 WARN  unwrapped dismissableRunInSubprocess(), falling back to blocking in-process run
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
error: ICC Profiles not supported in NO_ICC build
warning: could not load ICC profile in JPEG image
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
08/13/22-18:13:32 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-18:13:32 INFO  Inhibiting user input
dhcpcd[17840]: sending signal ARLM to pid 17755
dhcpcd[17840]: waiting for pid 17755 to exit
dhcpcd[17866]: dhcpcd not running
08/13/22-18:13:33 INFO  Kobo Wi-Fi: disabling Wi-Fi
dhcpcd[17879]: dhcpcd not running
Selected interface 'eth0'
OK
08/13/22-18:13:49 INFO  Kobo suspend: going to sleep . . .
08/13/22-18:13:49 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-18:13:51 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-18:13:51 INFO  Kobo suspend: synced FS
08/13/22-18:13:51 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-18:45:40 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-18:45:40 INFO  Kobo resume: clean up after wakeup
08/13/22-18:45:40 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-18:45:40 INFO  Restoring user input handling
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
08/13/22-18:59:35 INFO  Kobo Wi-Fi: enabling Wi-Fi
[enable-wifi.sh] Closed fd 14 -> /dev/urandom
[obtain-ip.sh] Closed fd 14 -> /dev/urandom
dhcpcd[17983]: dhcpcd not running
dhcpcd[17990]: version 6.6.6 starting
dhcpcd[17990]: eth0: using /sys hwaddr: 58:b0:d4:35:64:fc
dhcpcd[17990]: eth0: executing `/libexec/dhcpcd-run-hooks' PREINIT
dhcpcd[17990]: eth0: executing `/libexec/dhcpcd-run-hooks' CARRIER
dhcpcd[17990]: eth0: delaying IPv4 for 0.9 seconds
dhcpcd[17990]: eth0: using ClientID 01:58:b0:d4:35:64:fc
dhcpcd[17990]: eth0: soliciting a DHCP lease
dhcpcd[17990]: eth0: sending DISCOVER (xid 0xc85b3c31), next in 3.4 seconds
dhcpcd[17990]: eth0: offered 192.168.1.11 from 192.168.1.1
dhcpcd[17990]: eth0: sending REQUEST (xid 0xc85b3c31), next in 4.0 seconds
dhcpcd[17990]: eth0: acknowledged 192.168.1.11 from 192.168.1.1
dhcpcd[17990]: eth0: leased 192.168.1.11 for 86400 seconds
dhcpcd[17990]: eth0: renew in 43200 seconds, rebind in 75600 seconds
dhcpcd[17990]: eth0: writing lease `/var/db/dhcpcd-eth0-Kiwi.lease'
dhcpcd[17990]: eth0: adding IP address 192.168.1.11/24
dhcpcd[17990]: eth0: adding route to 192.168.1.0/24
dhcpcd[17990]: eth0: adding default route via 192.168.1.1
dhcpcd[17990]: eth0: executing `/libexec/dhcpcd-run-hooks' BOUND
dhcpcd[17990]: forking to background
dhcpcd[17990]: forked to background, child pid 18006
08/13/22-18:59:43 WARN  unwrapped dismissableRunInSubprocess(), falling back to blocking in-process run
g_mkdir failed: No such file or directory
error: ICC Profiles not supported in NO_ICC build
warning: could not load ICC profile in JPEG image
error: ICC Profiles not supported in NO_ICC build
warning: could not load ICC profile in JPEG image
08/13/22-19:31:22 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-19:31:22 INFO  Inhibiting user input
dhcpcd[18100]: sending signal ARLM to pid 18006
dhcpcd[18100]: waiting for pid 18006 to exit
dhcpcd[18126]: dhcpcd not running
08/13/22-19:31:23 INFO  Kobo Wi-Fi: disabling Wi-Fi
dhcpcd[18139]: dhcpcd not running
Selected interface 'eth0'
OK
08/13/22-19:31:39 INFO  Kobo suspend: going to sleep . . .
08/13/22-19:31:39 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-19:31:41 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-19:31:41 INFO  Kobo suspend: synced FS
08/13/22-19:31:41 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-19:38:39 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-19:38:39 INFO  Kobo resume: clean up after wakeup
08/13/22-19:38:39 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-19:38:39 INFO  Restoring user input handling
08/13/22-19:53:11 INFO  WakeupMgr: scheduling wakeup in 259200
08/13/22-19:53:11 INFO  Inhibiting user input
08/13/22-19:53:26 INFO  Kobo suspend: going to sleep . . .
08/13/22-19:53:26 INFO  Kobo suspend: successfully asked the kernel to put subsystems to sleep
08/13/22-19:53:28 INFO  Kobo suspend: waited for 2s because of reasons...
08/13/22-19:53:28 INFO  Kobo suspend: synced FS
08/13/22-19:53:28 INFO  Kobo suspend: asking for a suspend to RAM . . .
08/13/22-20:55:53 INFO  Kobo suspend: ZzZ ZzZ ZzZ... And woke up!
08/13/22-20:55:53 INFO  Kobo resume: clean up after wakeup
08/13/22-20:55:53 INFO  Kobo resume: successfully asked the kernel to resume subsystems
08/13/22-20:55:53 INFO  Restoring user input handling
g_mkdir failed: No such file or directory
08/13/22-21:08:38 INFO  Kobo Wi-Fi: enabling Wi-Fi
[enable-wifi.sh] Closed fd 14 -> /dev/urandom
[obtain-ip.sh] Closed fd 14 -> /dev/urandom
dhcpcd[18265]: dhcpcd not running
dhcpcd[18272]: version 6.6.6 starting
dhcpcd[18272]: eth0: using /sys hwaddr: 58:b0:d4:35:64:fc
dhcpcd[18272]: eth0: executing `/libexec/dhcpcd-run-hooks' PREINIT
dhcpcd[18272]: eth0: executing `/libexec/dhcpcd-run-hooks' CARRIER
dhcpcd[18272]: eth0: delaying IPv4 for 0.8 seconds
dhcpcd[18272]: eth0: using ClientID 01:58:b0:d4:35:64:fc
dhcpcd[18272]: eth0: soliciting a DHCP lease
dhcpcd[18272]: eth0: sending DISCOVER (xid 0xe8e43b38), next in 4.3 seconds
dhcpcd[18272]: eth0: offered 192.168.1.11 from 192.168.1.1
dhcpcd[18272]: eth0: sending REQUEST (xid 0xe8e43b38), next in 3.9 seconds
dhcpcd[18272]: eth0: acknowledged 192.168.1.11 from 192.168.1.1
dhcpcd[18272]: eth0: leased 192.168.1.11 for 86400 seconds
dhcpcd[18272]: eth0: renew in 43200 seconds, rebind in 75600 seconds
dhcpcd[18272]: eth0: writing lease `/var/db/dhcpcd-eth0-Kiwi.lease'
dhcpcd[18272]: eth0: adding IP address 192.168.1.11/24
dhcpcd[18272]: eth0: adding route to 192.168.1.0/24
dhcpcd[18272]: eth0: adding default route via 192.168.1.1
dhcpcd[18272]: eth0: executing `/libexec/dhcpcd-run-hooks' BOUND
dhcpcd[18272]: forking to background
dhcpcd[18272]: forked to background, child pid 18288
08/13/22-21:08:46 WARN  unwrapped dismissableRunInSubprocess(), falling back to blocking in-process run
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
g_mkdir failed: No such file or directory
./luajit: plugins/autowarmth.koplugin/main.lua:310: attempt to compare nil with number
stack traceback:
	plugins/autowarmth.koplugin/main.lua:310: in function 'scheduleNextWarmthChange'
	plugins/autowarmth.koplugin/main.lua:374: in function 'action'
	frontend/ui/uimanager.lua:1197: in function '_checkTasks'
	frontend/ui/uimanager.lua:1627: in function 'handleInput'
	frontend/ui/uimanager.lua:1744: in function 'run'
	./reader.lua:333: in main chunk
	[C]: at 0x000140bd
!!!!
Uh oh, something went awry... (Crash n°1: 08/13/22 @ 21:41:18)
Running FW 4.12.12111 on Linux 3.0.35+ (#5030 PREEMPT Wed Oct 25 10:25:24 CST 2017)
Attempting to restart KOReader . . .
!!!!
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 08/13/22-21:41:33
 [*] Version: v2022.07

ffi.load: blitbuffer
08/13/22-21:41:34 INFO  initializing for device Kobo_star
08/13/22-21:41:34 INFO  framebuffer resolution: {
    ["w"] = 758,
    ["h"] = 1024,
}
[ko-input] Forked off fake event generator (pid: 18420)
08/13/22-21:41:34 WARN  Device:getKeyRepeat: EVIOCGREP ioctl failed: Function not implemented
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
ffi.load: libs/libutf8proc.so.2
PING 192.168.1.1 (192.168.1.1): 56 data bytes
64 bytes from 192.168.1.1: seq=0 ttl=64 time=21.750 ms

--- 192.168.1.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 21.750/21.750/21.750 ms
PING 192.168.1.1 (192.168.1.1): 56 data bytes
64 bytes from 192.168.1.1: seq=0 ttl=64 time=2.006 ms

--- 192.168.1.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 2.006/2.006/2.006 ms
08/13/22-21:41:35 INFO  Loading plugins from directory: plugins
ffi.load: sqlite3
08/13/22-21:41:35 ERROR Failed to initialize autowarmth plugin:  plugins/autowarmth.koplugin/main.lua:310: attempt to compare nil with number
08/13/22-21:45:01 INFO  quitting uimanager

```
</details>

**Comment by Frenzie** [comment]
@zwim is this what you recently fixed?

**Comment by zwim** [comment]
Yes, this was fixed with #9411.

@symphorien you can install the latest nightly, to solve the crash.

**Comment by symphorien** [comment]
yes the latest nightly solves the crash
